api/main.go

```diff
@@ -48,37 +48,44 @@ func main() {
 	app.Version = Version
 	app.Flags = config.APIFlags()
 	app.Action = func(c *cli.Context) error {
-		cfg := config.NewContextAPIConfig(c)
-		if err := cfg.Validate(); err != nil {
+		if err := config.ValidateAPIContext(c); err != nil {
 			return err
 		}
 
 		logger := logging.NewLogWriter(c.Bool("debug"))
 		log.SetOutput(logger)
 
 		awsConfig := defaults.Get().Config
-		staticCreds := credentials.NewStaticCredentials(cfg.AccessKey(), cfg.SecretKey(), "")
+		staticCreds := credentials.NewStaticCredentials(
+			c.String(config.FlagAWSAccessKey.GetName()),
+			c.String(config.FlagAWSSecretKey.GetName()),
+			"")
+
 		awsConfig.WithCredentials(staticCreds)
-		awsConfig.WithRegion(cfg.Region())
-		awsConfig.WithMaxRetries(cfg.MaxRetries())
+		awsConfig.WithRegion(c.String(config.FlagAWSRegion.GetName()))
+		awsConfig.WithMaxRetries(config.DefaultMaxRetries)
 		session := session.New(awsConfig)
 
-		delay := c.Duration(config.FLAG_AWS_TIME_BETWEEN_REQUESTS)
+		delay := c.Duration(config.FlagAWSRequestDelay.GetName())
 		ticker := time.Tick(delay)
 		session.Handlers.Send.PushBack(func(r *request.Request) {
 			<-ticker
 		})
 
 		client := awsclient.NewClient(session)
-		tagStore := tag.NewDynamoStore(session, cfg.DynamoTagTable())
-		jobStore := job.NewDynamoStore(session, cfg.DynamoJobTable())
-
-		adminProvider := aws.NewAdminProvider(client, tagStore, cfg)
-		deployProvider := aws.NewDeployProvider(client, tagStore, cfg)
-		environmentProvider := aws.NewEnvironmentProvider(client, tagStore, cfg)
-		loadBalancerProvider := aws.NewLoadBalancerProvider(client, tagStore, cfg)
-		serviceProvider := aws.NewServiceProvider(client, tagStore, cfg)
-		taskProvider := aws.NewTaskProvider(client, tagStore, cfg)
+		tagStore := tag.NewDynamoStore(session, c.String(config.FlagAWSTagTable.GetName()))
+		jobStore := job.NewDynamoStore(session, c.String(config.FlagAWSJobTable.GetName()))
+
+		if err := addAPIEntityTags(tagStore); err != nil {
```

Comment:
This appears to be the only reference to `addAPIEntityTags()` and to me looks to be undefined elsewhere in the set of changes you've made.

Reply:
bad merge on my part. fixed.